src/components/Data/Documents/DocumentListViewColumn.vue

```diff
@@ -0,0 +1,293 @@
+                <autocomplete
+                  class="ListViewColumnInput"
+                  placeholder="Add column"
+                  :items="mappingArray"
+                  :value="newCustomField || ''"
+                  inputClass="ListViewColumnInput"
+                  @autocomplete::change="attribute => { newCustomField = attribute; addCustomField(); }"
+                  :notifyChange="false"
+                />
+              </div>
+            </div>
+          </form>
+        </th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr v-for="(doc, kk) in documents" :key="kk">
+        <td class="actions">
+          <dropdown
+            class="DocumentBoxItem-actions"
+            :id="`document-dropdown-${doc.id}`"
+            >
+            <li>
+              <a
+                v-bind:class="{'disabled': !canEdit}"
+                v-title="{active: !canDelete, title: 'You are not allowed to edit this document'}"
+                @click="editDocument(doc.id)"
+                >
+                Edit
+              </a>
+            </li>
+            <li class="divider"></li>
+            <li>
+              <a
+                v-bind:class="{'disabled': !canDelete}"
+                v-title="{active: !canDelete, title: 'You are not allowed to delete this document'}"
+                @click="deleteDocument(doc.id)"
+                >
+                Delete
+              </a>
+            </li>
+          </dropdown>
+        </td>
+        <td class="DocumentColumnItem">
+          {{doc.id}}
+        </td>
+        <td v-for="(attr, k) in customFields" :key="k" class="DocumentColumnItem">
+          <span v-if="parseDocument(attr, doc).isObject" class="relative">
+            <a href="#" @click.prevent="toggleJsonFormatter(attr+doc.id)">{{parseDocument(attr, doc).value}} ...</a>
+            <pre :ref="attr+doc.id" class="DocumentListViewColumn-jsonFormatter" style="visibility: hidden;" v-json-formatter="{content: parseDocument(attr, doc).realValue, open: true}"></pre>
+          </span>
+          <span v-else>
+            {{parseDocument(attr, doc).value}}
+          </span>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script>
+import Dropdown from '../../Materialize/Dropdown'
+import Title from '../../../directives/title.directive'
+import JsonFormatter from '../../../directives/json-formatter.directive'
+import {
+  canEditDocument,
+  canDeleteDocument
+} from '../../../services/userAuthorization'
+import Autocomplete from '../../Common/Autocomplete'
+import _ from 'lodash'
+
+export default {
+  name: 'DocumentListViewColumn',
+  props: {
+    documents: Array,
+    index: String,
+    collection: String,
+    value: [Object, String, Array],
+    mapping: Object
+  },
+  directives: {
+    Title,
+    JsonFormatter
+  },
+  components: {
+    Dropdown,
+    Autocomplete
+  },
+  mounted () {
+    const columnsConfig = JSON.parse(localStorage.getItem('columnViewConfig') || '{}')
```

Comment:
I'm not sure but I think if you are in "private navigation" on your browser, it's the `sessionStorage` that is filled instead of `localstorage`. It can be good to have a fallback to sessionStorage before the empty object :thinking:

Reply:
It's on purpose, I want to have a brand new environment when I browse in private.